aws-rds-dbinstance/src/main/java/software/amazon/rds/dbinstance/Translator.java

```diff
@@ -437,12 +436,18 @@ public static ModifyDbInstanceRequest modifyDbInstanceAfterCreateRequestV12(fina
                 .iops(model.getIops())
                 .masterUserPassword(model.getMasterUserPassword())
                 .preferredBackupWindow(model.getPreferredBackupWindow())
-                .preferredMaintenanceWindow(model.getPreferredMaintenanceWindow())
-                .build();
+                .preferredMaintenanceWindow(model.getPreferredMaintenanceWindow());
+
+        if (STORAGE_TYPE_GP3.equalsIgnoreCase(model.getStorageType())) {
+            builder.storageThroughput(model.getStorageThroughput());
+            builder.iops(model.getIops());
+            builder.storageType(model.getStorageType());
+        }
+        return builder.build();
```

Comment:
This is an ancient DBInstance configuration that would not support the newest storage type. We can omit it here.